persist/fs/files.go

```diff
@@ -53,6 +53,9 @@ func openFiles(opener fileOpener, fds map[string]**os.File) error {
 
 func closeFiles(fds ...*os.File) error {
 	for _, fd := range fds {
+		if fd == nil {
```

Comment:
Hm, why did we have to do this?  Seems like masking someone passing a `nil` fd here means we miss closing a file we explicitly asked to be closed but we messed up and passed nil instead.  This is one of those "I think I'd prefer to crash" pieces of code potentially.